GoMeet/src/main/java/com/google/sps/servlets/UpdateLocationServlet.java

```diff
@@ -0,0 +1,47 @@
+package main.java.com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import main.java.com.google.sps.data.Location;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.jsoup.Jsoup;
+import org.jsoup.safety.Whitelist;
+
+/** Handles fetching and saving location data. */
+@WebServlet("/update-location-data")
+public class UpdateLocationServlet extends HttpServlet {
```

Comment:
Optional suggestion: since this servlet will only update the vote (in fact, it will only increment the vote by 1), maybe consider naming it with a more specific name (e.g. VoteLocationServlet), unless you are considering to extend the functionality of this servlet in the future so that it will update more things than just increment the votes.

Reply:
One of my goal was to delete locations when voting reaches zero, I think that will be handled in this servlet too.